configure.ac

```diff
@@ -8,6 +8,26 @@ AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE([1.11 dist-bzip2 foreign -Wall -Werror subdir-objects])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
+AC_CANONICAL_HOST
+
+macos=0
+linux=0
+
+case $host_os in
+*darwin*)
+	macos=1
+	;;
+*linux*)
+	linux=1
+	;;
+*)
+	AC_MSG_ERROR([libfabric only builds on Linux & OS X])
+	;;
+esac
+
+AM_CONDITIONAL([MACOS], [test "x$macos" = "x1"])
+AM_CONDITIONAL([LINUX], [test "x$linux" = "x1"])
```

Comment:
Trivial note: I'd change these tests to be of the form:

```
test $macos -eq 1
```

because you *know* that they will be either 0 or 1, so you can use the numeric comparison (-eq and -ne and friends) rather than the string comparison ("" and =).  This has two (minor) benefits:

* the numeric comparison is apparently slightly more efficient (not really important here, but...)
* you don't have to use the "x" modifier to ensure that the string is not empty